airflow/providers/common/sql/hooks/sql.py

```diff
@@ -234,7 +266,42 @@ def run(
         """
         Runs a command or a list of commands. Pass a list of sql
         statements to the sql parameter to get them to execute
-        sequentially
+        sequentially.
+
+        The method will return either single query results (typically list of rows) or list of those results
+        where each element in the list are results of one of the queries (typically list of list of rows :D)
+
+        For compatibility reasons, the behaviour of the DBAPIHook is somewhat confusing.
+        In cases, when multiple queries are run, the return values will be an iterable (list) of results
+        - one for each query. However, in certain cases, when single query is run - the results will be just
+        the results of that query without wrapping the results in a list.
+
+        The cases when single query results are returned without wrapping them in a list are when:
+
+        a) sql is string and last_statement is True (regardless what split_statement value is)
+        b) sql is string and split_statement is False
+
+        In all other cases, the results are wrapped in a list, even if there is only one statement to process:
+
+        a) always when sql is iterable (regardless what last_statement value is)
+        b) when sql is string, split_statement is True and last_statement is False
```

Comment:
I'm not sure I understand a?
string is also iterable. so a always happens thus b will never be checked? 

Also, can you clarify the handling for when user is doing:

`SQLExecuteQueryOperator(..., sql=["SELECT 1"])`
vs
`SQLExecuteQueryOperator(..., sql="SELECT 1")`

In that case the are equivalent but from the description i'm not sure if handled the same?

Reply:
Ah yeah. Mental shortcut. "Iterable of string statements" now.

Reply:
Fixed